go/libraries/doltcore/sqle/dfunctions/hashof.go

```diff
@@ -101,20 +101,38 @@ func (t *HashOf) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 	return h.String(), nil
 }
 
-func getBranchInsensitive(ctx context.Context, branchName string, ddb *doltdb.DoltDB) (br ref.DoltRef, err error) {
+func getRefInsensitive(ctx context.Context, refName string, ddb *doltdb.DoltDB) (ref.DoltRef, error) {
```

Comment:
You changed the semantics here, to include other kinds of refs (not just branches). The existing call sites will not be pleased. You should add a new method rather than change this one.

Reply:
the only call site is merge. should a branch not be able to merge the root of a tag/head ref?